src/components/SearchBar/SearchBar.tsx

```diff
@@ -0,0 +1,31 @@
+import * as S from './SearchBar.style';
+import { ChangeEvent, SyntheticEvent } from 'react';
+import SEARCH_ICON from '@assets/icons/search.svg';
+import CLOSE from '@assets/icons/close.svg';
+
+interface Props {
+  value?: string;
+  onSearch?: (e: SyntheticEvent) => void;
+  onChange?: (e: ChangeEvent<HTMLInputElement>) => void;
+  onReset?: () => void;
+}
+
+function SearchBar({ value, onSearch, onChange, onReset }: Props) {
+  return (
+    <S.Form onSubmit={onSearch}>
```

Comment:
onSearch가 optional props라면 함수 실행을 할때에도 optional chaining 해주셔야해요.
```suggestion
    <S.Form onSubmit={()=> { onSearch?.()} }>
```